src/StackExchange.Redis/Message.cs

```diff
@@ -198,6 +201,18 @@ public bool IsAdmin
 
         public bool IsAsking => (Flags & AskingFlag) != 0;
 
+        public bool IsHighIntegrity => _highIntegrityChecksum != 0;
+        public int HighIntegrityChecksum => _highIntegrityChecksum;
+
+        internal void WithHighIntegrity(Random rng)
+        {
+            // create a required value if needed; avoid sentinel zero
+            while (_highIntegrityChecksum is 0)
```

Comment:
If we wrap around, will need to handle 0 in the sequence version

Reply:
done